src/app.jsx

```diff
@@ -73,12 +71,13 @@ class Application extends React.Component {
         this._asyncRequestImages = utils.varlinkCall(utils.PODMAN, "io.podman.ListImages")
                 .then(reply => {
                     this._asyncRequestImages = null;
-                    this.setState({ imagesMeta: reply.images });
-                    this.state.imagesMeta.map((img) => {
+                    let imagesMeta = reply.images || [];
+                    imagesMeta.map((img) => {
                         utils.varlinkCall(utils.PODMAN, "io.podman.InspectImage", {name: img.id})
                                 .then(reply => {
-                                    const temp_imgs = this.state.images;
-                                    temp_imgs.push(JSON.parse(reply.image));
+                                    let temp_imgs = [];
+                                    Object.keys(this.state.images).filter(id => { temp_imgs[id] = this.state.images[id] });
```

Comment:
This does't filter anything. I think the intention is to fill the `temp_imgs` dictionary, so please use `map()` in these cases. However, this transforms a dictionary (`this.state.images`) to a list, and back into a dictionary. At first (and also second) it looks like the whole effect of this line is just to copy the `this.state.images` dictionary, and then the next line changes one item.

This should rather be done with 

    this.setState({images: Object.assign({}, this.state.images, {img.id: JSON.parse(reply.image)})

or, if you want to use the l33t spread operator, you can do the copy and update like this:

    this.setState({images: {...this.state.images, img.id: JSON.parse(reply.image)})